controllers/local_db_statefulset.go

```diff
@@ -262,7 +263,7 @@ func (r *BackstageReconciler) cleanupLocalDbResources(ctx context.Context, backs
 		},
 	}
 	if _, err := r.cleanupResource(ctx, secret, backstage); err != nil {
-		return err
+		return fmt.Errorf("failed to delete generated database secret, reason: %s", err)
```

Comment:
General suggestion (but can be handled consistently in a separate PR): could we wrap original errors instead, so that we don't lose the underlying errors?

```suggestion
		return fmt.Errorf("failed to delete generated database secret, reason: %w", err)
```

Reply:
Lets handle this as a separate PR as additional thoughts are needed because the information we store in the status condition should be concise (ie, putting the full call stack is not a good idea), and on the other hand, we want the user to have as much information as possible.